### DISCUSSION TITLE
WARN  related org.keycloak.services.managers.AuthenticationManager

### DISCUSSION THREAD

**Discussion by mertcanakdeniz** [dasniko/keycloak-2fa-sms-authenticator#37] [Q&A]
Hey Dasniko, I  appreciated  that your improvment .. 
When I deployed your configuration on my Keycloak instance which is running on Docker so ı have deployed as .jar file in /opt/keycloak/providers/.   We are good till the here.  I can saw SMS OTP  options in actions. So i configured Browser with SMS and binded it as like your youtube video. But when i re-login to Keycloak with non admin user, sms code is didn't came. I can login directly.

Here the related log ..
2023-10-02 08:12:51,316 WARN  [org.keycloak.services.managers.AuthenticationManager] (executor-thread-76) Could not find configuration for Required Action mobile-number-ra, did you forget to register it?

We can discuss the issue, i will glad. 

Again thanks for your developments .

**Comment by dasniko** [comment]
As the log entry says... the required action `mobile-number-ra` could not be found.
Keycloak wants to save the RA at the user, as the user doesn't have an attribute `mobile_number` configured.
The corresponding code can be found here: https://github.com/dasniko/keycloak-2fa-sms-authenticator/blob/71c989e33df439ee3588b490477009b603630809/src/main/java/dasniko/keycloak/authenticator/SmsAuthenticator.java#L107-L112
Either, adjust the code to your requirements or deploy also the required action.

**Comment by mertcanakdeniz** [comment]
Thanks your reply. So I am seeing the RA on the user. 
But Keycloak does not have any related configuration behind the mobile-number-ra action which is as you said. So i decided to overriding with your demo extention repo. Here : 
https://github.com/dasniko/keycloak-extensions-demo/tree/main/requiredaction/src/main/java/dasniko/keycloak/requiredaction
This can be solve the problem.  I  will give a feedback about the issue I' ll be done.  
![Screenshot from 2023-10-05 09-27-03](https://github.com/dasniko/keycloak-2fa-sms-authenticator/assets/76827927/3f0248e0-e19d-4b68-b5dd-fb98a05e60cc)
